### PR TITLE
http server declaration

### PR DESCRIPTION
**Pull request by gyzerok**
Added missing http server and createServer declarations

### CODE REVIEW

File: lib/node.js [hunk 1]
```diff
@@ -551,6 +551,15 @@ declare module "fs" {
 }
 
 declare module "http" {
+  declare class Server extends events$EventEmitter {
+    listen(port: number, hostname?: string, backlog: number, callback?: Function): Server;
```
Comment (gabelevi) [inline lib/node.js]: I think backlog should be optional too?

File: lib/node.js [hunk 2]
```diff
@@ -551,6 +551,15 @@ declare module "fs" {
...
+    listen(path: string, callback?: Function): Server;
+    listen(handle: Object, callback?: Function): Server;
+    close(callback?: Function): Server;
+    address(): { port: number; fmaily: string; address: string; };
```
Comment (gabelevi) [inline lib/node.js]: I don't see this in [http.Server](https://nodejs.org/api/http.html#http_class_http_server) but I do see this in [net.Server](https://nodejs.org/api/net.html#net_server_address). Am I looking at the wrong docs?

**Comment by gabelevi** [comment]
I have some small nits but this generally looks good to me!

**Comment by gyzerok** [comment]
@gabelevi I guess here is a bug in node documentation since I do know that http.Server has address method. Check TypeScript [declarations](https://github.com/borisyankov/DefinitelyTyped/blob/master/node/node.d.ts#L347).

**Pull request closed without merging**